engine/src/main/java/org/terasology/rendering/nui/widgets/UISlider.java

```diff
@@ -173,14 +171,39 @@ public String getMode() {
             return DISABLED_MODE;
         }
 
-        if (active) {
+        if (active || (TabbingManager.focusedWidget != null && TabbingManager.focusedWidget.equals(this))) {
             return ACTIVE_MODE;
         } else if (tickerListener.isMouseOver()) {
             return HOVER_MODE;
         }
         return DEFAULT_MODE;
     }
 
+    private void changeValue(float delta) {
+        float newValue = TeraMath.clamp(getValue() + delta, 0, getRange() + getMinimum());
+        setValue(newValue);
+    }
+
+    @Override
+    public boolean onKeyEvent(NUIKeyEvent event) {
+        if (event.isDown() && TabbingManager.focusedWidget.equals(this)) {
+            int keyId = event.getKey().getId();
+            if (keyId == Keyboard.KeyId.RIGHT || keyId == Keyboard.KeyId.UP) {
+                this.changeValue(getIncrement());
+                return true;
+            } else if (keyId == Keyboard.KeyId.LEFT || keyId == Keyboard.KeyId.DOWN) {
+                this.changeValue(-1 * getIncrement());
+                return true;
+            }
+        }
+        return super.onKeyEvent(event);
+    }
+
+    @Override
+    public void onMouseWheelEvent(MouseWheelEvent event) {
+        event.consume();
```

Comment:
Just curious of the logic behind consuming this event. I don't know the code well enough to work it out through a review

Reply:
As far as I understand, consume in supposed to stop the event from propagating after this widget receives it, right? I put it there because I wanted to end that event as soon as it started and not interfere with mouse button release.